File: backend/api/models/event.py
from django.db import models

TYPES = ["journal submission", "academic conference", "funded project"]
TYPE_CHOICES = [(TYPES[i], str(i)) for i in range(len(TYPES))]


class Event(models.Model):
    """
    Event model
    """
    title = models.CharField(max_length=100, blank=False)
    description = models.CharField(max_length=200, blank=True, default='')
    deadline = models.DateField(blank=False)
    date = models.DateField(blank=False)
    event_type = models.CharField(
        choices=TYPE_CHOICES, blank=False, max_length=50)
    url = models.URLField(blank=True, default='')

    class Meta:
        ordering = ['date']

File: android/app/src/main/java/com/bounswe2020group3/paperlayer/project/data/Event.kt
package com.bounswe2020group3.paperlayer.project.data

import com.squareup.moshi.Json

data class Event (
        @field:Json(name = "id")
        var id: Int,
        @field:Json(name = "title")
        var title: String,
        @field:Json(name = "description")
        var description: String,
        @field:Json(name = "deadline")
        var deadline: String,
        @field:Json(name = "date")
        var date: String,
        @field:Json(name = "event_type")
        var eventType: String,
        @field:Json(name = "url")
        var url: String
)